tests/remotes/webdav.py

```diff
@@ -17,19 +16,25 @@
 
 
 class Webdav(Base, WebDAVURLInfo):
+    _DIR_PATH = None
+
     @staticmethod
     def get_url(port):  # pylint: disable=arguments-differ
         return f"webdav://localhost:{port}"
 
     def mkdir(self, mode=0o777, parents=False, exist_ok=False):
-        assert parents
-        makedirs(self.path, exist_ok=exist_ok)
+        self.dir_path.mkdir(parents=parents, exist_ok=True)
 
     def write_bytes(self, contents):
-        Path(self.path).write_bytes(contents)
+        self.dir_path.write_bytes(contents)
 
     def write_text(self, contents, encoding=None, errors=None):
-        Path(self.path).write_text(contents, encoding=encoding, errors=errors)
+        self.dir_path.write_text(contents, encoding=encoding, errors=errors)
+
+    @property
+    def dir_path(self):
```

Comment:
The `self.path` points to the root `/data` rather then a relative position in the given temporary directory, since it tries to obtain it from the url (`webdav://stuff:stufff/stuff`). So this points to the temporary directory location

Reply:
Could we do it the way FakeHadoopFileSystem does it? `Webdav._DIR_PATH = Path(directory)` is awkward.

Reply:
@efiop one problem with initializing this in the init rather than as a class var is that, we won't be inheriting it in the sub paths and then there would be no point unless I am missing something

Reply:
@isidentical Hm, not sure I follow. We have `_hdfs_root` in hdfs, that we use as a "server root", which seems to have the same meaning as `_DIR_PATH`.

Reply:
So are you proposing to put it as a global-var?

Reply:
@isidentical the root? Yeah, if we can. Notice how we are then using uuid to isolate. So we emulate having one real server that way.

Reply:
I think it should work now, though we don't clean the directory though I am not sure whether we need an `atexit.register()` call or just let it handled by the interpreter close.